### PR TITLE
SG-208 react native codebase detect

### PR DESCRIPTION
**Pull request by augsteyer**
## Description

Checks for the version of our App base & adds a  body class based on that. The class is used to add a margin-top only on older version of our app base. The tests provide 4 different scenarios for old Ios device & Android device. The new app treats ios & android equally, so only one test is necessary. Last tests checks non-SG app call.

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I updated the CHANGELOG.md

### CODE REVIEW

**Comment by augsteyer** [comment]
Not sure how to proceed with dependency issue in the audit.

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/) and this
 - ability to skip advanced price export when calling `get_items`, config called `skip_advanced_price_export`
 - webCheckout login route for guests
 - webCheckout will close browser when an item is added to cart on desktop site (inApp)
+- CSS body tag that differentiates between R.Native (`.is-sg-codebase-v2`) & old Swift/Java based App (`.is-sg-codebase-v1`)
```
Comment (Rekhyt) [inline CHANGELOG.md:13]:
```suggestion

### Fixed
- layout issues in web checkout on Shopgate apps based on the newer code base (> version 11.0.0)

```

File: src/Backend/SgateShopgatePlugin/Bootstrap.php [hunk 2]
```diff
@@ -1488,12 +1471,41 @@ public function onFrontendCustom(Enlight_Event_EventArgs $args)
...
+        $isNative = $this->isNativeBase($userAgent);
+        $view->assign('sgIsNativeBase', $isNative);
+
         $customCss = Shopware()->Config()->getByNamespace('SgateShopgatePlugin', 'SGATE_CUSTOM_CSS');
         $view->assign('sgCustomCss', $customCss);
     }
 
+    /**
+     * Native SG App should have a Codebase variable with
+     * a version higher than 11
```
Comment (Rekhyt) [inline src/Backend/SgateShopgatePlugin/Bootstrap.php:1496]:
```suggestion
     * React Native base SG app should have a "Codebase" variable with version >= 11.0.0 in its user agent.
```

File: src/Backend/SgateShopgatePlugin/Bootstrap.php [hunk 3]
```diff
@@ -1488,12 +1471,41 @@ public function onFrontendCustom(Enlight_Event_EventArgs $args)
...
+     *
+     * @param string $userAgent
+     * @return bool
+     */
+    private function isNativeBase($userAgent)
```
Comment (Rekhyt) [inline src/Backend/SgateShopgatePlugin/Bootstrap.php:1501]:
```suggestion
    private function isReactNativeBase($userAgent)
```

File: src/Backend/SgateShopgatePlugin/Bootstrap.php [hunk 4]
```diff
@@ -1488,12 +1471,41 @@ public function onFrontendCustom(Enlight_Event_EventArgs $args)
...
+     *
+     * @return void
+     */
+    private function assignCommonData(Enlight_View_Default $view, Enlight_Event_EventArgs $args)
+    {
         $view->assign('sgSessionId', Shopware()->Session()->offsetGet('sessionId'));
 
+        $userAgent = $args->getSubject()->Request()->getHeader('user-agent');
+        $isNative = $this->isNativeBase($userAgent);
+        $view->assign('sgIsNativeBase', $isNative);
```
Comment (Rekhyt) [inline src/Backend/SgateShopgatePlugin/Bootstrap.php:1488]:
```suggestion
        $isReactNative = $this->isReactNativeBase($userAgent);
        $view->assign('sgIsReactNativeBase', $isReactNative);
```

File: src/Backend/SgateShopgatePlugin/Views/frontend/index/index.tpl [hunk 5]
```diff
@@ -7,7 +7,7 @@
         <html class="no-js" lang="{s name='IndexXmlLang'}{/s}" itemscope="itemscope" itemtype="http://schema.org/WebPage" data-disallow-pull-to-refresh="1">
     {/if}
 {/block}
-
+{block name="frontend_index_body_classes"}{$smarty.block.parent}{if $sgWebCheckout} {strip}is-sg-codebase-{if $sgIsNativeBase}v2{else}v1{/if}{/strip}{/if}{/block}
```
Comment (Rekhyt) [inline src/Backend/SgateShopgatePlugin/Views/frontend/index/index.tpl:10]:
```suggestion
{block name="frontend_index_body_classes"}{$smarty.block.parent}{if $sgWebCheckout} {strip}is-sg-codebase-{if $sgIsReactNativeBase}v2{else}v1{/if}{/strip}{/if}{/block}
```

File: tests/Postman/collection.json [hunk 6]
```diff
@@ -1,2021 +1,2279 @@
...
-                    "articles"
-                  ]
-                }
-              },
-              "response": []
-            }
-          ]
-        }
-      ]
+    "info": {
```
Comment (Rekhyt) [inline tests/Postman/collection.json:2]: 2 spaces plz 🙈

**Review by Rekhyt — CHANGES_REQUESTED** [review]
Thank you for the PR!

I made a few remarks, mainly about the use of "native" without "react", being a bit ambiguous imo because both code bases are native apps. Was there a reason to avoid the name of the framework  in use?

File: tests/Postman/collection.json (hunk 6 above)
Comment (augsteyer) [reply tests/Postman/collection.json:2]: I could, but um, I am just following the SW5 .editorconfig policy: https://github.com/shopware/shopware/blob/trunk/.editorconfig#L11

I could overwrite to a 2 if that's what you wish, submit a file with our repo with just:
```
[*.json]
indent_size = 2
```
What you think @Rekhyt ?

Comment (Rekhyt) [reply tests/Postman/collection.json:2]: I think it says 2 spaces in the file you linked: https://github.com/shopware/shopware/blob/trunk/.editorconfig#L19

Plus, it was 2 before, so the diff is all over the place now. Not unsovable but if we don't have to switch to 4 we shouldn't.

Comment (augsteyer) [reply tests/Postman/collection.json]: Ye, my bad, wrong repo (that's sw6), was looking at local config & made a mistake in repo link https://github.com/shopware5/shopware/blob/5.7/.editorconfig#L11

No problem though, I created a config rewrite.

Comment (Rekhyt) [reply tests/Postman/collection.json]: That's cool, thank you 👍

**Review by Rekhyt — APPROVED** [review]

**Pull request merged**